src/Symfony/Component/Messenger/MessageRecorderInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+interface MessageRecorderInterface
+{
+    /**
+     * Fetch recorded messages.
+     *
+     * @return object[]
+     */
+    public function recordedMessages(): array;
```

Comment:
Why not naming this function `fetch()` instead of `recordedMessages()`, as you already name below the method `record()` (without *messages* at the end of the name) ?

Same thing for `eraseMessages()`.

Reply:
I took the names from SimpleBus. But you are correct, your naming seams easier